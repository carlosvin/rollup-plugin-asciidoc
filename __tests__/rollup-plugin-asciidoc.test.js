const asciidocPlugin = require('../src/rollup-plugin-asciidoc')
const rollup = require('rollup')
const path = require('path')
const Module = require('module')

function requireFromString(code) {
  const opts = {}
  const filename = 'test_require_from_string.js'

  opts.appendPaths = opts.appendPaths || []
  opts.prependPaths = opts.prependPaths || []

  if (typeof code !== 'string') {
    throw new Error('code must be a string, not ' + typeof code)
  }

  var paths = Module._nodeModulePaths(path.dirname(filename))

  var parent = module.parent
  var m = new Module(filename, parent)
  m.filename = filename
  m._compile(code, filename)

  var exports = m.exports
  parent &&
    parent.children &&
    parent.children.splice(parent.children.indexOf(m), 1)

  return exports
}

process.chdir(__dirname)

const bundleFileAndGetCode = async rollupConfig => {
  const bundle = await rollup.rollup(rollupConfig)

  const { output } = await bundle.generate({ format: 'cjs' })

  const [{ code }] = output
  return code
}

it('returns a module for the asciidoc file', async () => {
  const code = await bundleFileAndGetCode({
    input: 'fixtures/test.adoc',
    plugins: [asciidocPlugin()],
  })

  const requiredModule = requireFromString(code)

  expect(requiredModule.html).toMatchSnapshot()
  expect(requiredModule.metadata).toEqual({
    layout: 'post',
    title: 'Avoiding recursive useEffect hooks in React',
    intro: expect.any(String),
  })
  expect(requiredModule.filename).toEqual('test.adoc')
  expect(requiredModule.path).toEqual(path.resolve(path.join(__dirname, 'fixtures/test.adoc')))
})
