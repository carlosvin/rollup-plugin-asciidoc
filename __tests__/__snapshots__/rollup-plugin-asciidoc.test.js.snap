// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a module for the asciidoc file 1`] = `
Object {
  "filename": "test.adoc",
  "html": "<div class=\\"paragraph\\">
<p>It’s fair to say that React 16.8 and the introduction of
<a href=\\"https://reactjs.org/docs/hooks-intro.html\\">hooks</a> has really changed how
we write React. Hooks are one of those APIs that make you realise the
flaws of the previous approach <em>after</em> you stop using it. I remember
being very skeptical of hooks when they were first released, not
thinking that the previous class based design had many flaws, but I’ve
since come to realise I was very wrong, and hooks are a vast improvement
on how we build React components. If you’re interested in comparing the
old vs the new, I wrote a <a href=\\"/refactoring-to-react-hooks/\\">blog post
refactoring a component to use hooks</a> that offers a nice comparison.</p>
</div>
<div class=\\"listingblock\\">
<div class=\\"content\\">
<pre class=\\"highlightjs highlight\\"><code data-lang=\\"python\\" class=\\"language-python hljs\\">def main():
    pass</code></pre>
</div>
</div>",
  "metadata": Object {
    "appendix-caption": "Appendix",
    "appendix-refsig": "Appendix",
    "asciidoctor": "",
    "asciidoctor-version": "2.0.10",
    "asdf": "made up attr",
    "attribute-missing": "skip",
    "attribute-undefined": "drop-line",
    "author": "Doc Writer",
    "authorcount": 1,
    "authorinitials": "DW",
    "authors": "Doc Writer",
    "backend": "html5",
    "backend-html5": "",
    "backend-html5-doctype-article": "",
    "basebackend": "html",
    "basebackend-html": "",
    "basebackend-html-doctype-article": "",
    "caution-caption": "Caution",
    "chapter-refsig": "Chapter",
    "docdate": "2020-04-12",
    "docdatetime": "2020-04-12 01:43:09 +0200",
    "docdir": "",
    "docfile": "test.adoc",
    "docfilesuffix": ".adoc",
    "docname": "test",
    "doctime": "01:43:09 +0200",
    "doctitle": "My Document's Title",
    "doctype": "article",
    "doctype-article": "",
    "docyear": "2020",
    "email": "doc.writer@asciidoctor.org",
    "embedded": "",
    "example-caption": "Example",
    "figure-caption": "Figure",
    "filetype": "html",
    "filetype-html": "",
    "firstname": "Doc",
    "htmlsyntax": "html",
    "iconfont-remote": "",
    "iconsdir": "./images/icons",
    "important-caption": "Important",
    "intro": "A short post today about an easy tactic to avoid your useEffect calls becoming recursive when setting state.",
    "last-update-label": "Last updated",
    "lastname": "Writer",
    "layout": "post",
    "linkcss": "",
    "localdate": undefined,
    "localdatetime": undefined,
    "localtime": undefined,
    "localyear": undefined,
    "max-attribute-value-size": 4096,
    "max-include-depth": 64,
    "note-caption": "Note",
    "notitle": "",
    "outfilesuffix": ".html",
    "part-refsig": "Part",
    "prewrap": "",
    "revdate": "2014-01-01",
    "revnumber": "1.0",
    "safe-mode-level": 20,
    "safe-mode-name": "secure",
    "safe-mode-secure": "",
    "sectids": "",
    "section-refsig": "Section",
    "stylesdir": ".",
    "stylesheet": "",
    "table-caption": "Table",
    "tip-caption": "Tip",
    "title": "'Avoiding recursive useEffect hooks in React'",
    "toc-placement": "auto",
    "toc-title": "Table of Contents",
    "untitled-label": "Untitled",
    "user-home": ".",
    "version-label": "Version",
    "warning-caption": "Warning",
    "webfonts": "",
  },
  "path": "/home/carlos/workspace/rollup-plugin-asciidoc/__tests__/fixtures/test.adoc",
}
`;
